doc/source/tutorials/add_model.rst

```diff
@@ -0,0 +1,327 @@
+.. _tutorial-add-model:
+
+:octicon:`ruby` Add Your Own Model
+==================================
+
+.. dropdown:: What you will learn
+    :icon: multi-select
+    :animate: fade-in
+
+    * How to add a new model to an existing model family
+    * How to register model configurations (architectures)
+    * How to create asset cards for your models
+    * How to verify your model integration works correctly
+
+.. dropdown:: Prerequisites
+    :icon: multi-select
+    :animate: fade-in
+
+    * Get familiar with fairseq2 basics (:ref:`basics-design-philosophy`)
+    * Understanding of fairseq2 assets system (:ref:`basics-assets`)
+    * Ensure you have fairseq2 installed
+
+Overview
+--------
+
+fairseq2 uses a model system that makes it easy to add support for new models.
+There are two main scenarios:
+
+1. **Adding a new model to an existing family** (most common) - When you want to add a new size or variant of an existing model architecture
+2. **Creating an entirely new model family** (advanced) - When you need to implement a completely new model architecture
+
+This tutorial focuses on the first scenario, which covers 95% of use cases. For the second scenario, refer to the existing model family implementations as reference.
+
+Understanding Model Families
+----------------------------
+
+In fairseq2, a **model family** groups related model architectures that share the same underlying implementation but differ in size or configuration. For example:
+
+- **Qwen family**: ``qwen25_3b``, ``qwen25_3b``, ``qwen25_14b``, ``qwen3_8b``, etc.
+- **LLaMA family**: ``llama3_8b``, ``llama3_70b``, ``llama3_2_1b``, etc.
+- **Mistral family**: ``mistral_7b``, ``mistral_8x7b``, etc.
+
+Each family consists of:
+
+- **Model configurations** (architectures): Define structural parameters (layers, dimensions, etc.)
+- **Asset cards**: YAML files specifying download locations and metadata
+- **Model implementation**: The actual PyTorch model code and loading logic
+
+Step-by-Step Guide: Adding a Model to Existing Family
+-----------------------------------------------------
+
+Let's walk through adding ``qwen25_3b_instruct`` to the existing Qwen family.
+
+Step 1: Add Model Architecture Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Model architectures are defined as configuration presets that specify the structural parameters of your model.
```

Comment:
If we are aiming for external users, I believe it would make more sense to show how to do this without needing to touch fairseq2 code base.

```python
def setup_my_fairseq2_extension(container: DependencyContainer) -> None:
  arch = ConfigRegistrar(container, QwenConfig)

  @arch("my_qwen_arch")
  def my_qwen_arch() -> QwenConfig:
     config = QwenConfig()
     ...
     return config
```

Either register `setup_my_fairseq2_extension` if this code resides in a Python package, or use `init_fairseq2(extras=setup_my_fairseq2_extension)` if this lives inside a plain Python script.